### PR TITLE
prow: make GC intervals configurable

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/3408

### CODE REVIEW

File: prow/cmd/sinker/main.go [hunk 1]
```diff
@@ -59,20 +57,21 @@ func main() {
 	pkc := kc.Namespace(configAgent.Config().PodNamespace)
 
 	// Clean now and regularly from now on.
-	clean(kc, pkc)
-	t := time.Tick(period)
+	clean(kc, pkc, configAgent)
+	t := time.Tick(configAgent.Config().Sinker.ResyncPeriod)
```
Comment (spxtr) [inline prow/cmd/sinker/main.go]: This won't be reread if we update the `ResyncPeriod`.  I suggest rereading the period every loop and manually waiting the appropriate amount of time.

Comment (kargakis) [inline prow/cmd/sinker/main.go]: Updated

**Comment by kargakis** [comment]
Rebased because of a conflict in the Makefile

**Pull request merged**